app/components/home.jsx

```diff
@@ -1,34 +1,34 @@
 'use client';
-import { useState } from 'react';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
 import InfiniteScroll from 'react-infinite-scroller';
-import { fetchPokemons } from '@/api/fetch-pokemons';
 import styles from './home.module.css';
 import { PokemonItem } from './pokemon-item/pokemon-item';
+import { fetchPokemonsThunk } from '@/redux/features/pokemon/service';
+import {
+  getPokemon,
+  getHasMore,
+  getIsLoading,
+} from '@/redux/features/pokemon/selector';
+import { setPokemonList } from '@/redux/features/pokemon/slice';
+import { useAppDispatch } from '@/redux/store';
 
 export const HomePage = ({ initialPokemonDetails }) => {
-  const limit = 10;
-  const [pokemon, setPokemon] = useState(initialPokemonDetails ?? []);
-  const [hasMore, setHasMore] = useState(true);
-  const [offset, setOffset] = useState(initialPokemonDetails?.length ?? 0);
-  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useAppDispatch();
+  const pokemon = useSelector(getPokemon);
+  const hasMore = useSelector(getHasMore);
+  const isLoading = useSelector(getIsLoading);
+
+  useEffect(() => {
+    if (initialPokemonDetails.length > 0) {
```

Comment:
Use the condition `!pokemon.length` instead.
This will ensure this happens only when the pokemon list is empty.

Reply:
shouldn't this be pokemon.length, since it is supposed to happen when it is not empty